enterprise/server/usage/usage.go

```diff
@@ -292,200 +292,97 @@ func (ut *tracker) flushCounts(ctx context.Context, groupID string, c collection
 			pk.Region,
 			pk.GroupID,
 			pk.PeriodStartUsec,
-		).Rows()
-		if err != nil {
+		).Take(&tables.Usage{}).Error
+		if err != nil && !db.IsRecordNotFound(err) {
 			return err
 		}
-
-		schema, err := db.TableSchema(tx, &tables.Usage{})
-		if err != nil {
-			return status.WrapError(err, "failed to get usage table schema")
-		}
-
-		existingRowCount := 0
-		for rows.Next() {
-			existingRowCount++
-			fields := map[string]any{}
-			if err := tx.ScanRows(rows, &fields); err != nil {
-				return err
-			}
-
-			unsupportedField := ""
-			var unsupportedFieldValue any
-			for f, v := range fields {
-				if v == nil || v == "" {
-					continue
-				}
-				if _, ok := schema.FieldsByDBName[f]; !ok {
-					unsupportedField = f
-					unsupportedFieldValue = v
-					break
-				}
-			}
-			if unsupportedField != "" {
-				alert.UnexpectedEvent("usage_update_dropped", "Usage update transaction aborted since existing usage row contains unsupported column %q = %q (key = %+v)", unsupportedField, unsupportedFieldValue, pk)
-				return nil
-			}
-		}
-		if err := rows.Err(); err != nil {
-			return err
+		if err == nil {
+			log.Warningf("Usage update dropped since the row already exists; this should only happen if redis locking has failed.")
```

Comment:
do you want to make this an unexpected event?

I think we want to treat usage data failing as a pageable offense :)

Reply:
I was on the fence about this since (a) in this case, we're not actually 'dropping' usage (the warning is somewhat confusingly worded), we're just skipping an update because it's a dupe, and (b) I figured we don't want to treat the redis locking stuff as a hard dependency. But I guess it doesn't hurt to alert here, since if redis goes down we're probably getting alerted anyway :sweat_smile: